api/src/main/java/org/apache/iceberg/InputSplit.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.Collection;
+
+/**
+ * A scan task that may include partial input files, multiple input files or both.
+ *
+ * @param <T> the type of scan tasks
+ */
+public interface InputSplit<T extends ScanTask> extends ScanTask {
```

Comment:
This is a parent interface added to avoid breaking `CombinedScanTask`.

Reply:
I'm not sure about this name. We purposely didn't use the name `InputSplit` initially because we didn't want to conflict with Hadoop and other uses. Now, I'm afraid that it will duplicate those classes, and that it will be confusing to have both tasks and splits, with an unclear relationship between the two.

It also isn't clear why an `InputSplit` would have files, but a scan task would not. What about `CommonFileTask` or something similar?

Reply:
Oh, I tried at least 3 different names and went back and forth with them. Note this is a task that is usually referenced by an input split in query engines. The ideal name is `CombinedScanTask` as it is a task created by combining smaller tasks into a bigger one. However, we can't parameterize `CombinedScanTask` to keep compatibility.

I am afraid `CommonFileTask` is very generic and does not imply it may consist of multiple smaller tasks. Maybe, something with `Packed`? We need a reasonable synonym for `CombinedScanTask`. Any ideas are welcome.

I am also not sure `files()` is very appropriate. I inherited it from `CombinedScanTask` but it makes more sense there.

Reply:
Maybe, something like `CompoundScanTask<T>`?
Or maybe drop it at all and define `CombinedChangelogScanTask` that would extend `ScanTask` directly?

Reply:
I'll think more about this but this is the biggest open question at the moment. Ideas are welcome.

Reply:
I went with `ScanTaskGroup` for now. It acts simply like a group of tasks now. Let me know what you think.